app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -65,7 +65,7 @@ if ! [ -x "$(command -v subscription-manager)" ] ; then
 else
   echo "subscription-manager is already installed!"
   <% if @subman_setup_scenario == 'registration' -%>
-    $PKG_MANAGER_UPGRADE subscription-manager > /dev/null 2>&1
+    $PKG_MANAGER_UPGRADE --setopt=obsoletes=0 subscription-manager > /dev/null 2>&1
```

Comment:
```suggestion
    $PKG_MANAGER_UPGRADE --exclude=rhn-client-tools subscription-manager > /dev/null 2>&1
```
This approach seems to me a bit more directed. Using obsoletes=0 may cause unintended effects down the road if sub-man or any of its deps use obsoletes.

Reply:
We already have this bit here:
https://github.com/theforeman/foreman/blob/44160c0db58a954c92c3dc912841a9acea4d74bf/app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb#L32-L33

Right now I haven't looked at it in detail, but this PR implies that the original suggestion was insufficient.

Reply:
The suggestion by @bocekm does indeed work more reliable. I'll address that in the PR.

Reply:
@ekohl, as far as I recall from my testing some time back removing `rhn-client-tools` before installing sub-man on OL 7 is not sufficient - yum says that sub-man is obsoleted by rhn-client-tools and goes ahead and installs rhn-client-tools instead of sub-man.

Reply:
@danmyway, perhaps you should fix the lines @ekohl mentioned above as well.